### ISSUE TITLE
Installation issue: r@4.2.2

### ISSUE THREAD

**Issue by odoublewen** [LLNL/spack#35684]
### Steps to reproduce the issue

```console
$ spack spec -I r@4.2.2
Input spec
--------------------------------
 -   r@4.2.2

Concretized
--------------------------------
 -   r@4.2.2%gcc@4.8.5~X~external-lapack~memory_profiling~rmath build_system=autotools arch=linux-centos7-haswell
[+]      ^bzip2@1.0.8%gcc@4.8.5~debug~pic+shared build_system=generic arch=linux-centos7-haswell
[+]          ^diffutils@3.8%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]              ^libiconv@1.17%gcc@4.8.5 build_system=autotools libs=shared,static arch=linux-centos7-haswell
[+]      ^curl@7.87.0%gcc@4.8.5~gssapi~ldap+libidn2~librtmp~libssh~libssh2~nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-centos7-haswell
[+]          ^libidn2@2.3.4%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]              ^libunistring@1.1%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]          ^openssl@1.1.1t%gcc@4.8.5~docs~shared build_system=generic certs=mozilla arch=linux-centos7-haswell
[+]              ^ca-certificates-mozilla@2023-01-10%gcc@4.8.5 build_system=generic arch=linux-centos7-haswell
[+]          ^pkgconf@1.8.0%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]      ^icu4c@66.1%gcc@4.8.5 build_system=autotools cxxstd=11 arch=linux-centos7-haswell
[+]          ^python@3.10.8%gcc@4.8.5+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-centos7-haswell
[+]              ^expat@2.5.0%gcc@4.8.5+libbsd build_system=autotools arch=linux-centos7-haswell
[+]                  ^libbsd@0.11.7%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]                      ^libmd@1.0.4%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]              ^gdbm@1.23%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]              ^libffi@3.4.4%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]              ^libxcrypt@4.4.33%gcc@4.8.5~obsolete_api build_system=autotools arch=linux-centos7-haswell
[+]              ^sqlite@3.40.1%gcc@4.8.5+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-centos7-haswell
[+]              ^util-linux-uuid@2.36.2%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]      ^ncurses@6.4%gcc@4.8.5~symlinks+termlib abi=none build_system=autotools arch=linux-centos7-haswell
[+]      ^openjdk@11.0.17_8%gcc@4.8.5 build_system=generic arch=linux-centos7-haswell
[+]      ^pcre2@10.42%gcc@4.8.5~jit+multibyte build_system=autotools arch=linux-centos7-haswell
[+]      ^readline@8.2%gcc@4.8.5 build_system=autotools patches=bbf97f1 arch=linux-centos7-haswell
[+]      ^texinfo@7.0%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]          ^gettext@0.21.1%gcc@4.8.5+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-centos7-haswell
[+]              ^libxml2@2.10.3%gcc@4.8.5~python build_system=autotools arch=linux-centos7-haswell
[+]              ^tar@1.34%gcc@4.8.5 build_system=autotools zip=pigz arch=linux-centos7-haswell
[+]                  ^pigz@2.7%gcc@4.8.5 build_system=makefile arch=linux-centos7-haswell
[+]                  ^zstd@1.5.2%gcc@4.8.5+programs build_system=makefile compression=none libs=shared,static arch=linux-centos7-haswell
[+]          ^perl@5.36.0%gcc@4.8.5+cpanm+open+shared+threads build_system=generic arch=linux-centos7-haswell
[+]              ^berkeley-db@18.1.40%gcc@4.8.5+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-centos7-haswell
[+]      ^which@2.21%gcc@4.8.5 build_system=autotools arch=linux-centos7-haswell
[+]      ^xz@5.4.1%gcc@4.8.5~pic build_system=autotools libs=shared,static arch=linux-centos7-haswell
[+]      ^zlib@1.2.13%gcc@4.8.5+optimize+pic+shared build_system=makefile arch=linux-centos7-haswell
```

### Error message

```console
==> r: Executing phase: 'build'                                                                                                                       [92/489]
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4' 'V=1'

3 errors found in build log:
     1851    JNI cpp flags    : -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
     1852    JNI linker flags : -L$(JAVA_HOME)/lib/server -ljvm
     1853    Updating Java configuration in /tmp/owen/spack-stage/spack-stage-r-4.2.2-hsyfp53vjbzohgfki7ldhvrzfc7bpyjx/spack-src
     1854    Done.
     1855    
     1856    make[1]: Leaving directory `/tmp/owen/spack-stage/spack-stage-r-4.2.2-hsyfp53vjbzohgfki7ldhvrzfc7bpyjx/spack-src'
  >> 1857    Error: compiling TeX file 'reshape.tex' failed with message:
     1858    Running 'texi2dvi' on 'reshape.tex' failed.
     1859    LaTeX errors:
     1860    ! LaTeX Error: File `grfext.sty' not found.
     1861    
     1862    Type X to quit or <RETURN> to proceed,
     1863    or enter new name. (Default extension: sty)

     ...

     1866    <read *>
     1867    
     1868    l.179   \RequirePackage{grfext}\relax
     1869                                         ^^M
     1870    !  ==> Fatal error occurred, no output PDF file produced!
     1871    Execution halted
  >> 1872    make[1]: *** [vignettes] Error 1
     1873    make[1]: Leaving directory `/tmp/owen/spack-stage/spack-stage-r-4.2.2-hsyfp53vjbzohgfki7ldhvrzfc7bpyjx/spack-src/src/library'
  >> 1874    make: *** [vignettes] Error 2
     1875    make: *** Waiting for unfinished jobs....
     1876    creating doc/NEWS.pdf
     1877    make[2]: Entering directory `/tmp/owen/spack-stage/spack-stage-r-4.2.2-hsyfp53vjbzohgfki7ldhvrzfc7bpyjx/spack-src/doc/manual'
     1878    creating doc/manual/R-FAQ.html
     1879    creating doc/manual/version.texi
     1880    creating doc/manual/R-admin.html

See build log for details:
  /tmp/owen/spack-stage/spack-stage-r-4.2.2-hsyfp53vjbzohgfki7ldhvrzfc7bpyjx/spack-build-out.txt
```

### Information on your system

* **Spack:** 0.20.0.dev0 (d286146c641ce826eeb67089bd3e24912406e51b)
* **Python:** 3.6.8
* **Platform:** linux-centos7-skylake_avx512
* **Concretizer:** clingo

Mentioning @glennpj as listed maintainer of `r`

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/10828902/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/10828903/spack-build-out.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by odoublewen** [comment]
FWIW, I replicated this error on a completely different system with a more modern OS.

* **Spack:** 0.20.0.dev0 (c4311a250a151228e76b7b29e32f9497d79b1c0c)
* **Python:** 3.9.2
* **Platform:** linux-debian11-excavator
* **Concretizer:** clingo

**Comment by glennpj** [comment]
@odoublewen This is because the host does not have a full enough TeX distribution, and there is not currently a way to tell the R build to _not_ use TeX if it detects part of it. I think there are three options.

1. You could install TeX from your distribution packages and the R build will use it.
2. Build texlive with Spack and load it before building R.
3. Add a build dependency on texlive in the r package recipe.
`depends_on("texlive", type="build")`

At one time I had a PR to add the dependency on texlive but it could not pass the CI tests due to timeouts and so I abandoned that.

**Comment by odoublewen** [comment]
Thank you.  Yesterday, my colleague and I finally figured out that including `texlive` as a dep could make it work.  I strongly favor a soluion that is doesn't depend on distro packages -- I think that's the ethos of spack.  But yes, as I think you are alluding to, adding `texlive` adds a ton of (I think mostly not needed) dependencies so it's a shame to do it that way.

Maybe this is more a question/issue for the texlive package.  It looks like the `texlive@live` version offers the ability to set a "scheme" to get a more minimal install.  But as I'm sure you know, the `@live` version is marked as deprecated in spack.  (Yet, it appears to be the [promoted installation method](https://www.tug.org/texlive/quickinstall.html) on the project homepage.)

Or maybe the solution is to make a variant in the R package  (`r@4.2.2+texlive`) so that folks who need to can do it that way, and folks who don't, needn't.   I still wonder if using a minimal install of texlive could work.  e.g., could something like this work:  `r@4.2.2+texlive^texlive@live+schema=minimal`?  (I *think* that's how one would specify both a variant `texlive` and also specify the spec for the `texlive` dependency?)

Of course, for the above to work, we'd have to define a `texlive` variant in the `r` package.

cc: @birnbera @hartzell

**Comment by hartzell** [comment]
[Note, I'm on the same side of a corporate firewall as Owen...]
@odoublewen -- @glennpj [mentioned a few comments up](https://github.com/spack/spack/issues/35684#issuecomment-1448770899) that building R documentation is triggered by the R install process discovering *some* TeX bits but having those TeX bits not be sufficient.    Do you know what TeX bits it's tripping over?

**Comment by odoublewen** [comment]
@hartzell  The missing bits are latex style files, and I believe R just needs them to build its docs/vignettes.  

The error message above says `LaTeX Error: File 'grfext.sty' not found.`  But I've also seen other `*.sty` files come up in that error message.  I'm not sure if the variability is due to my playing with the spec, or if there's a race/threading situation in the build.

**Comment by odoublewen** [comment]
I can no longer replicate this error, installing r@4.3.0, with Spack 0.21.0.  The R package now installs correctly as-is, without the need to bring in the (substantial) burden of building texlive.  I notice the R's configure script outputs:
```
configure: WARNING: you cannot build PDF versions of the R manuals
configure: WARNING: you cannot build PDF versions of vignettes and help pages
```
while spack is configuring it, so apparently R already anticipated skipping the PDF manual builds if the config detected it wasn't possible.  I looked at the R source and it appears that this same warning message was also in place back in 4.2.2, so really I am not sure why it was failing when I opened this issue back in March.

Anyway, I am happy close this issue.. bigger fish to fry...
